programs/escrow/src/lib.rs

```diff
@@ -121,8 +147,17 @@ pub mod escrow {
 pub struct EscrowAccount {
     pub giver_key: Pubkey,
     pub taker_key: Pubkey,
-    pub giver_token_account: Pubkey,
+    pub giver_token_account: Pubkey, // Don't think we actually need to store this. As long as the future giver_token_account is owned by the giver_key, should be ok?.
```

Comment:
the purpose was to ensure the funds get refunded to the original account. perhaps unnecessary restriction